integration/src/main/java/ai/djl/integration/tests/nn/BlockFactoryTest.java

```diff
@@ -0,0 +1,114 @@
+import java.io.IOException;
+import org.testng.annotations.Test;
+
+public class BlockFactoryTest {
+
+    @Test
+    public void testBlockLoadingSaving()
+            throws IOException, ModelNotFoundException, MalformedModelException,
+                    TranslateException {
+        TestBlockFactory factory = new TestBlockFactory();
+        Model model = factory.constructBlock();
+        try (NDManager manager = NDManager.newBaseManager()) {
+            Shape inputShape = new Shape(1, 3, 32, 32);
+            Block block = model.getBlock();
+            block.setInitializer(Initializer.ONES);
+            block.initialize(manager, DataType.FLOAT32, inputShape);
+            ByteArrayOutputStream os = new ByteArrayOutputStream();
+            block.saveParameters(new DataOutputStream(os));
+            ByteArrayInputStream bis = new ByteArrayInputStream(os.toByteArray());
+            Block newBlock = factory.getBlock(manager);
+            newBlock.loadParameters(manager, new DataInputStream(bis));
+            try (Model test = Model.newInstance("test")) {
+                test.setBlock(newBlock);
+                try (Predictor<NDList, NDList> predOrigin =
+                                model.newPredictor(new NoopTranslator());
+                        Predictor<NDList, NDList> predDest =
+                                test.newPredictor(new NoopTranslator())) {
+                    NDList input = new NDList(manager.ones(new Shape(1, 3, 32, 32)));
+                    NDList originOut = predOrigin.predict(input);
+                    NDList destOut = predDest.predict(input);
+                    Assertions.assertAlmostEquals(originOut, destOut);
+                }
+            }
+        }
+    }
+
+    static class TestBlockFactory implements BlockFactory {
+
+        @Override
+        public Block getBlock(NDManager manager) {
+            SequentialBlock newBlock = new SequentialBlock();
+            newBlock.add(SymbolBlock.newInstance(manager));
+            newBlock.add(Blocks.batchFlattenBlock());
+            newBlock.add(Linear.builder().setUnits(10).build());
+            return newBlock;
+        }
+
+        public Model constructBlock()
+                throws MalformedModelException, ModelNotFoundException, IOException {
+            String name = Engine.getInstance().getEngineName();
+            Criteria.Builder<Image, Classifications> builder =
+                    Criteria.builder()
+                            .optApplication(Application.CV.IMAGE_CLASSIFICATION)
+                            .setTypes(Image.class, Classifications.class)
+                            .optProgress(new ProgressBar())
+                            .optArtifactId("resnet")
+                            .optEngine(name)
+                            .optGroupId("ai.djl." + name.toLowerCase())
+                            .optFilter("layers", "50");
+            Model model = ModelZoo.loadModel(builder.build());
```

Comment:
In this case it loads the model with just name, since you don't specify any url/file?

Reply:
No, it does looking for the model stored in DJL model zoo.